src/server/http/setup_connection.js

```diff
@@ -48,7 +48,7 @@ export default function (kbnServer, server, config) {
   });
 
   server.ext('onRequest', function (req, reply) {
-    if (req.raw.req.socket.encrypted) {
+    if (!req.raw.req.socket || req.raw.req.socket.encrypted) {
```

Comment:
I think your description would make a great comment here, to explain the context:

```javascript
// A request sent through a HapiJS .inject() doesn't have a socket associated with the request,
// which causes a failure.
if (!req.raw.req.socket || req.raw.req.socket.encrypted) {
```